### PR TITLE
Use Case: Generate Printable Map PDF

### PR DESCRIPTION
**Pull request by NurramoX**
This use case refers to #1 about printing maps onto a PDF and provide meta information in the header or footer 

## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

File: doc/usecases/map_to_pdf.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+## Scenarios
+
+- **Precondition:**
+ - The user has opened the app and navigated to the desired map view.
+- **Main success scenario:**
+ - The app captures a screenshot of the current map view and generates a PDF file.
+ - The app adds a header/footer to the PDF containing the current time, map timestamp, GPS coordinates, and scale.
+ - The user receives the generated PDF file for printing or saving.
+- **Alternative scenario:**
+ - The user selects custom options e.g. for the header/footer, such as changing the format or choosing which metadata to include.
```
Comment (markus2330) [inline doc/usecases/map_to_pdf.md:21]:
```suggestion
```

File: doc/usecases/map_to_pdf.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Use Case: Generate Printable Map PDF
```
Comment (markus2330) [inline doc/usecases/map_to_pdf.md:1]: Please rename:

```suggestion
# Use Case: Map to PDF
```

File: doc/usecases/map_to_pdf.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+## Summary
+
+- **Scope:** Map View
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** Users can generate a PDF file containing a screenshot of their current map view with a header/footer displaying relevant metadata, such as current time, map timestamp, GPS coordinates, and scale.
```
Comment (markus2330) [inline doc/usecases/map_to_pdf.md:8]:
```suggestion
- **Brief:** Users can generate a PDF file of the map.
```

File: doc/usecases/map_to_pdf.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+- **Status:** Draft
```
Comment (markus2330) [inline doc/usecases/map_to_pdf.md:9]:
```suggestion
- **Status:** Assigned
```

File: doc/usecases/map_to_pdf.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ - The user has opened the app and navigated to the desired map view.
+- **Main success scenario:**
+ - The app captures a screenshot of the current map view and generates a PDF file.
+ - The app adds a header/footer to the PDF containing the current time, map timestamp, GPS coordinates, and scale.
+ - The user receives the generated PDF file for printing or saving.
+- **Alternative scenario:**
+ - The user selects custom options e.g. for the header/footer, such as changing the format or choosing which metadata to include.
+- **Error scenario:**
+ - The app encounters an issue while generating the PDF file, such as insufficient storage space or an unexpected error.
+- **Postconition:**
```
Comment (markus2330) [inline doc/usecases/map_to_pdf.md:24]:
```suggestion
- **Postcondition:**
```

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great work, only very minor issues.

File: doc/usecases/map_to_pdf.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+- **Assignee:** Ramzan
+
+## Scenarios
+
+- **Precondition:**
+ - The user has opened the app and navigated to the desired map view.
+- **Main success scenario:**
+ - The app captures a screenshot of the current map view and generates a PDF file.
+ - The app adds a header/footer to the PDF containing the current time, map timestamp, GPS coordinates, and scale.
+ - The user receives the generated PDF file for printing or saving.
```
Comment (badnames) [inline doc/usecases/map_to_pdf.md:19]: How will this accomplished? 
I think the most reasonable thing would be an overview page with download links for each generated file.

**Review by badnames — APPROVED** [review]
Well done, just a two minor things I noticed.

File: doc/usecases/map_to_pdf.md (hunk 6 above)
Comment (markus2330) [reply doc/usecases/map_to_pdf.md:19]: I would prefer if we can save it to Nextcloud. Then we could also easily implement features like automatic deletion https://apps.nextcloud.com/apps/files_retention

Btw. thanks for the reviews!

**Comment by markus2330** [comment]
Now part of #366

**Pull request closed without merging**